src/modules/navigator/mission.cpp

```diff
@@ -244,6 +244,12 @@ void Mission::setActiveMissionItems()
 			mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
 		}
 
+		// prevent lateral guidance from loitering at a waypoint as part of a mission landing if the altitude
+		// is not achieved.
+		if (isLanding() && _mission_item.nav_cmd == NAV_CMD_WAYPOINT) {
```

Comment:
I don't think this is safe. The DO_LAND_START waypoint can be attached to any normal waypoint, which then changes the altitude handling of every following waypoint.

Reply:
@sfuhrer I don't quite understand. It will actually not change the behavior for the waypoint it's attached to (see definition of isLanding()) but yes, every position waypoint that comes after that. That's what we desire, no?

Reply:
Let's make an example: a MC user flies the drone in a complex environment. As he can't use the normal "go above and descend" RTL he plans a mission landing with 3 waypoints and then a land (e.g. following a narrow corridor). To mark the beginning of the RTL he sets a land start marker to the 4th to last waypoint. He expects that the drone will follow these 4 waypoints normally, including normal altitude behavior. 
What I'm basically going towards: it's not naturally clear that waypoints after the land_start marker have a different altitude behavior.

Reply:
@sfuhrer Does it ever make sense to plan a mission landing for MC? Also, MC does not even use the alt acceptance radius for the triplet. But it's true that navigator would accept the waypoint even if altitude is not reached (usually that does not happen for MC, since it can track any glide slope, but still it's a difference to before.)

Reply:
@sfuhrer Another idea is to define a new waypoint type which explicity says that altitude should not be enforced. Then the ground station could just use these waypoints and there is not confusion. While planning the mission the user could even select the altitude tracking behavior.